### PR TITLE
Icons hotfix

### PR DESCRIPTION
**Pull request by brunobar79**
## What changed (plus any additional context for devs)
- Handle raw data response properly from provider.call
- Fix logic on icon checks
- Backwards compatibility

## Screen recordings / screenshots

https://github.com/user-attachments/assets/55805daf-af1c-4012-aef3-7e32349b1ab5

## What to test
https://rainbowhaus.slack.com/archives/C0468CDBE75/p1734465096538089

### CODE REVIEW

File: src/featuresToUnlock/tokenGatedUtils.ts [hunk 1]
```diff
@@ -43,7 +43,10 @@ export const checkIfWalletsOwnNft = async (
 
     const data = iface.encodeFunctionData('areOwners(address[], address[])', [tokenAddresses, walletsToCheck]);
     const found = await p.call({ to: TOKEN_GATE_CHECKER_ADDRESS[network], data });
-    return found;
+    if (found === '0x0000000000000000000000000000000000000000000000000000000000000000') {
```
Comment (brunobar79) [inline src/featuresToUnlock/tokenGatedUtils.ts:46]: The response of provider.call is raw data (not a boolean 😅 )

File: src/featuresToUnlock/unlockableAppIconCheck.ts [hunk 2]
```diff
@@ -27,34 +27,38 @@ export const unlockableAppIconCheck = async (appIconKey: UnlockableAppIconKey, w
...
+      }
+      return allChecks;
+    });
+
+    const allPromises = promises.flat();
+    const results = await Promise.all(allPromises);
+
+    const found = results.some(result => !!result);
+    if (!found) {
+      unlockableAppIconStorage.set(appIconKey, false);
```
Comment (brunobar79) [inline src/featuresToUnlock/unlockableAppIconCheck.ts:60]: This is required to restore the proper state

**Review by derHowie — APPROVED** [review]

**Review by jinchung — APPROVED** [review]
🌮

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@bruno/icons-hotfix/e6fd8d8c4d2d51dff4d4b1ce1d4582dc39a72bd5.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@bruno/icons-hotfix/e6fd8d8c4d2d51dff4d4b1ce1d4582dc39a72bd5.ipa) for e6fd8d8c4d2d51dff4d4b1ce1d4582dc39a72bd5

**Pull request merged**